adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+import org.apache.hadoop.hbase.{ HBaseConfiguration, TableName }
+import org.apache.hadoop.hbase.io.compress.Compression.Algorithm
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.hbase.mapreduce.LoadIncrementalHFiles
+import org.apache.hadoop.fs.{ FSDataInputStream, FileSystem, Path }
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.rdd.variant.{ VariantContextRDD }
+import org.bdgenomics.adam.rich.RichVariant
+import org.bdgenomics.formats.avro._
+import org.bdgenomics.adam.models.{ ReferencePosition, ReferenceRegion, SequenceDictionary, VariantContext }
+import scala.io.Source
+import scala.collection.mutable.ListBuffer
+import scala.collection.JavaConverters._
+import sys.process._
+
+object HBaseFunctions {
+
+  /**
+   * HBase Spark Data Access Object
+   *
+   * In standard usage user will create a single instance of this object, and pass it as the first parameter
+   * when using the HBaseFunctions public API functions
+   *
+   * @param sc SparkContext
+   * @param inConf optional HBase Configuration parameter, only used if manually configuring HBase connection
+   * @param inConnection optional HBase Connection parameter, only used if manually configuring HBase connection
+   * @param inAdmin optional HBase Admin parameter, only used if manually configuring HBase connection
+   * @param inHBaseContext optional HBaseContext parameter, only used if manually configuring HBase connection
+   */
+
+  class HBaseSparkDAO(sc: SparkContext,
+                      inConf: Option[Configuration] = None,
+                      inConnection: Option[Connection] = None,
+                      inAdmin: Option[Admin] = None,
+                      inHBaseContext: Option[HBaseContext] = None) {
+
+    val conf = if (inConf.isDefined) inConf.get else HBaseConfiguration.create()
+    val connection = if (inConnection.isDefined) inConnection.get else ConnectionFactory.createConnection(conf)
+    val admin = if (inAdmin.isDefined) inAdmin.get else connection.getAdmin
+    val hbaseContext = if (inHBaseContext.isDefined) inHBaseContext.get else new HBaseContext(sc, conf)
+
+    def createTable(tableDescriptorMeta: HTableDescriptor) { admin.createTable(tableDescriptorMeta) }
+    def getTable(tableName: TableName): Table = { connection.getTable(tableName) }
+
+    def getHBaseRDD(tableName: org.apache.hadoop.hbase.TableName,
+                    scans: org.apache.hadoop.hbase.client.Scan): RDD[scala.Tuple2[org.apache.hadoop.hbase.io.ImmutableBytesWritable, org.apache.hadoop.hbase.client.Result]] = {
+      hbaseContext.hbaseRDD(tableName, scans)
+
+    }
+
+    def hbaseBulkLoad(genodata: RDD[(Array[Byte], List[(String, Array[Byte])])],
+                      hbaseTableName: String,
+                      flatMap: scala.Function1[(Array[Byte], List[(String, Array[Byte])]), scala.Iterator[scala.Tuple2[org.apache.hadoop.hbase.spark.KeyFamilyQualifier, scala.Array[scala.Byte]]]],
```

Comment:
change `scala.Function1[(Array[Byte], List[(String, Array[Byte])]), scala.Iterator[scala.Tuple2[org.apache.hadoop.hbase.spark.KeyFamilyQualifier, scala.Array[scala.Byte]]]]` to `((Array[Byte], List[(String, Array[Byte])]) => Iterator[(org.apache.hadoop.hbase.spark.KeyFamilyQualifier, Array[Byte])]`

Also, use short names, no?